src/mon/AuthMonitor.cc

```diff
@@ -1334,6 +1334,7 @@ bool AuthMonitor::valid_caps(
   if (type == "mon") {
     MonCap moncap;
     if (!moncap.parse(caps, out)) {
+      dout(20) << "Parsing MON caps failed. MON cap: " << caps << dendl;
```

Comment:
`out` would generally have the failure error string returned back to the CLI - were those not verbose enough for debugging. Esp. since the message logged here are pretty generic.

Reply:
It's better when logs too have an entry about failed parsing.

Reply:
IMO, it'll be a bit more helpful if the log message has details about _what_ failed. WDYT?

Reply:
Okay.  The needs the iterator that parsing uses. And that is internal to classes that parse.

Reply:
IMO, unless we additionally log the reason for parsing failure, this change does not add anything useful to debug.

Reply:
It's not useless because it's impossible to tell that parsing failed just by looking at logs. Such a message helps one to jump to a relevant part of the log and start investigating for the cause of failure from there.

Otherwise, logs is not very helpful as reader can't tell which messages were logged before the parsing failure.

Reply:
I'll leave it upto @rzarzynski to take a call for this.

Reply:
> It's not useless because it's impossible to tell that parsing failed just by looking at logs

I agree that a developer not-always has access to CLI's output; having the info in one place (in the log) helps, IMHO.